src/librustc_resolve/lib.rs

```diff
@@ -2440,6 +2442,21 @@ impl<'a, 'crateloader: 'a> Resolver<'a, 'crateloader> {
         self.ribs[TypeNS].pop();
     }
 
+    fn resolve_type(&mut self,
```

Comment:
I'd rename this to `resolve_type_definition_with_self_ty` or something more "obvious" - but why does it exists, couldn't you write this in-place?